code/headers/hardware_usart.hpp

```diff
@@ -29,6 +29,38 @@ namespace r2d2 {
     public:
         hardware_usart_c(unsigned int baudrate, uart_ports_c &usart_port);
 
+        /// console character output function
+        ///
+        /// This is the function used for console (ostream) output.
+        /// The embedded targets provide an implementation that writes
+        /// to the serial port.
+        ///
+        /// This definition is weak, which allows
+        /// an application to provide its own definition.
+        void uart_putc(char c);
+
+        /// @brief char output operator
+        ///
+        /// Although calling send_byte should do the exact same thing.
+        /// In practice useing this fuction is more stable
+        /// Especially when the values are repeated.
+        hardware_usart_c &operator<<(uint8_t byte) {
```

Comment:
Functions and should not be defined in the header, same goes for overloaded operators. sorry for not commenting on this earlier